[L-rrr] iP

L-rrrr opened the pull request:
### Shenhe

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

Shenhe frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~FAST~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is FREE!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

L-rrrr committed:
Change Duke.java to Shenhe.java and complete level 0

L-rrrr committed:
Add Shenhe's introduction, repeat function and bye function

L-rrrr committed:
Add add task and list task function

L-rrrr committed:
Add Task.java, mark as done, mark as undone and new list display

L-rrrr committed:
Add Todo.java, Deadline.java and Event.java

L-rrrr committed:
Add new exception classes and error handing in Shenhe.java

L-rrrr committed:
Add delete function in Shenhe.java

L-rrrr committed:
Add input.txt, EXPECTED.TXT and change runtest.bat

L-rrrr committed:
Add Storage.java and edit all task files

L-rrrr committed:
Merge branch 'branch-Level-7'

L-rrrr committed:
Add Date.Parser.java and edit all revelant files

L-rrrr committed:
Merge branch 'branch-Level-8'

L-rrrr committed:
Make all files more OOP

L-rrrr committed:
Merge branch 'A-MoreOOP'

L-rrrr committed:
Organise files into packages according to their functions

L-rrrr committed:
Merge branch 'add-gradle-support'

L-rrrr committed:
Add JUnit and some test files

L-rrrr committed:
Add javadoc in all files

L-rrrr committed:
Merge branch 'branch-A-JavaDoc'

L-rrrr committed:
Change shenhe.java so that it follows the standard format

L-rrrr committed:
Merge branch 'branch-A-CodingStandard'

L-rrrr committed:
Add search task by keywords

L-rrrr committed:
Merge branch 'branch-Level-9'

L-rrrr committed:
Add javadoc for findTask and showMatchingMessage method

TechRaj commented on src/main/java/shenhe/Storage.java:
```diff
@@ -0,0 +1,91 @@
+import shenhe.parser.Parser;
+import shenhe.task.Task;
+
+import java.io.*;
```
wildcard import

TechRaj commented on src/main/java/shenhe/command/DeleteCommand.java:
```diff
@@ -0,0 +1,73 @@
+     * @return {@code false} as the command does not terminate the application.
+     */
+    @Override
+    public boolean isExit() {
```
correct boolean method name. Lot of people miss out on this. good job!

TechRaj commented on src/main/java/shenhe/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,81 @@
+ * </p>
+ */
+public class DeadlineCommand extends Command {
+    private String userInput;
```
Make this final. Same for all commands inheriting from Command

TechRaj commented on src/main/java/shenhe/TaskList.java:
```diff
@@ -0,0 +1,111 @@
+ * </p>
+ */
+public class TaskList {
+    private List<Task> tasks;
```
make final

TechRaj reviewed (commented):
Good job! Just some nitpicks

L-rrrr committed:
Make all concrete command class and TaskList class final

L-rrrr committed:
Check coding style and modify all files

L-rrrr committed:
Merge branch 'A-CheckStyle'

Soumyadip-cmd commented on src/main/java/shenhe/parser/DateParser.java:
```diff
@@ -0,0 +1,61 @@
+     * @return A {@code LocalDateTime} object representing the parsed date and time, or {@code null} if the input
+     *         string does not match the expected format.
+     */
+    public static LocalDateTime parse(String dateTimeStr) {
```
Perhaps a more descriptive name here like parseDateTime?

Soumyadip-cmd commented on src/main/java/shenhe/parser/DateParser.java:
```diff
@@ -0,0 +1,61 @@
+     * @return A {@code LocalDateTime} object representing the parsed date and time, or {@code null} if the input
+     *         string does not match the expected format.
+     */
+    public static LocalDateTime parseFile(String dateTimeStr) {
```
Perhaps a more descriptive name?

Soumyadip-cmd commented on src/main/java/shenhe/parser/Parser.java:
```diff
@@ -0,0 +1,123 @@
+     *                   {@link InvalidListEnquiry} if the list command is invalid,
+     *                   {@link UnknownTaskException} if the command is not recognized.
+     */
+    public static Command parse(String userInput) throws Exception {
```
Would parseCommand be a better name here?

Soumyadip-cmd reviewed (commented):
Overall good naming of classes and methods. Included some suggestions for method names to improve understandability

L-rrrr committed:
Add GUI for the chatbox

L-rrrr committed:
Merge tag 'Level-10'

L-rrrr committed:
Change the parse function name to be more descriptive

L-rrrr committed:
Add assertions in Task, Parser, and DateParser

- Added assertions in the Task class to ensure that task descriptions are not null or empty, and task status is correctly initialized.
- Implemented assertions in the Parser class to validate assumptions about command parsing, ensuring that inputs conform to expected formats.
- Added assertions in the DateParser class to check that date and time strings match the expected patterns before parsing.
- These assertions help catch invalid inputs early, enhancing code reliability and preventing unexpected runtime errors.
- Updated test cases across all affected classes to account for new assertions and validate their behavior.

L-rrrr committed:
Make Storage.java adhere to coding standard

L-rrrr committed:
Improve search functionality to support partial and case-insensitive matches

The current search functionality in TaskList only finds tasks when the keyword matches the description exactly, including case sensitivity.
This approach limits the usability of the search feature, as users need to enter the exact match, making it difficult to find relevant tasks, especially with variations in case or incomplete keywords.
Update the findTasks method to perform case-insensitive comparisons and allow partial matches of keywords within task descriptions.
Using case-insensitive and substring checks enhances the search capability, making it more user-friendly and intuitive by accommodating a broader range of search inputs.
This change improves the overall user experience by making it easier to find tasks with less precise or varied input, aligning with common search expectations.

L-rrrr committed:
Merge pull request #1 from L-rrrr/A-Assertions

Add assertions

L-rrrr committed:
Merge branch 'master' into A-CodeQuality

L-rrrr committed:
Merge branch 'master' into BetterSearch

L-rrrr committed:
Merge pull request #2 from L-rrrr/A-CodeQuality

A code quality

L-rrrr committed:
Merge pull request #3 from L-rrrr/BetterSearch

Better search

L-rrrr committed:
Improve GUI design and add new JUnit tests

- Enhanced the GUI by resizing profile photos for better layout and aesthetics.
- Adjusted the background picture
- Refined the button hover and pressed states for better interactivity and responsiveness

- Added JUnit tests for TaskList and EmptyTaskDescriptionException:
  - Verified task retrieval, addition, deletion, and search functionalities in TaskList.
  - Implemented tests to ensure EmptyTaskDescriptionException is properly thrown and includes the correct error message.

Overall, this commit improves the user interface and strengthens the codebase with comprehensive unit tests.

L-rrrr committed:
Add README, improve UI and edit some responses

L-rrrr committed:
Add format in README
